### PR TITLE
[RPC] Allow in-wallet management of peer bans

### PR DESCRIPTION
**Pull request by Fuzzbawls**
Work in progress for now, This PR will be used to add in-wallet ban management capability that doesn't require a client restart. See individual commit messages for more details until this can be finalized.

- [x] Add setban/listbanned/clearbanned RPC Commands
- [x] Add disconnectnode RPC Command
- [x] Add persistent storage of ban list
- [x] Introduce the lightweight CScheduler class

### CODE REVIEW

File: src/net.cpp [hunk 1]
```diff
@@ -360,6 +360,15 @@ CNode* FindNode(const CNetAddr& ip)
     return NULL;
 }
 
+CNode* FindNode(const CSubNet& subNet)
+{
+    LOCK(cs_vNodes);
+    BOOST_FOREACH(CNode* pnode, vNodes)
```
Comment (presstab) [inline src/net.cpp]: I know this is from upstream, but we shouldn't ever add `BOOST_FOREACH` to the codebase and should replace it when we can with built in `for` range search in c++11. `for (CNode* pnode : vNodes)`

Comment (Fuzzbawls) [reply src/net.cpp]: yep, was planning on replacing the `BOOST_FOREACH`, just wanted to get this up ASAP for functionality testing.

**Comment by Fuzzbawls** [comment]
This is now fully ready for review, been running this for the past week or so without any issue across mainnet, testnet, and even partially during my early regtest testing.

**Comment by Fuzzbawls** [comment]
Rebased for use with UniValue

**Comment by Mrs-X** [comment]
Hmh, I've always liked this feature in my upstream wallets:
banui

Is this from upstream or did you implement something from scratch?

**Comment by Fuzzbawls** [comment]
This is from upstream, but doesn't include any UI ban management (that will be a separate PR).

This lays the foundation for ban management, as well as implements the management of bans via the RPC interface.

**Comment by Fuzzbawls** [comment]
@Mrs-X you left out the port in your `disconnectnode` test command. The argument is a full literal string, including port as it is entirely possible that there can be multiple peer connections to the same address, but different ports.

try `disconnectnode "45.77.233.27:51474"`

**Comment by Sieres** [comment]
ACK

**Pull request merged**